[2024-12-31T19:06:38.546+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_kafka_to_postgres_dag.submit_spark_job manual__2024-12-30T18:50:53.726705+00:00 [queued]>
[2024-12-31T19:06:38.580+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_kafka_to_postgres_dag.submit_spark_job manual__2024-12-30T18:50:53.726705+00:00 [queued]>
[2024-12-31T19:06:38.582+0000] {taskinstance.py:1308} INFO - Starting attempt 2 of 3
[2024-12-31T19:06:38.622+0000] {taskinstance.py:1327} INFO - Executing <Task(SparkSubmitOperator): submit_spark_job> on 2024-12-30 18:50:53.726705+00:00
[2024-12-31T19:06:38.636+0000] {standard_task_runner.py:57} INFO - Started process 56 to run task
[2024-12-31T19:06:38.642+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spark_kafka_to_postgres_dag', 'submit_spark_job', 'manual__2024-12-30T18:50:53.726705+00:00', '--job-id', '112', '--raw', '--subdir', 'DAGS_FOLDER/teams_kafka_to_spark.py', '--cfg-path', '/tmp/tmpvkvl2aqc']
[2024-12-31T19:06:38.648+0000] {standard_task_runner.py:85} INFO - Job 112: Subtask submit_spark_job
[2024-12-31T19:06:38.762+0000] {task_command.py:410} INFO - Running <TaskInstance: spark_kafka_to_postgres_dag.submit_spark_job manual__2024-12-30T18:50:53.726705+00:00 [running]> on host c646b2da3ab1
[2024-12-31T19:06:38.908+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_kafka_to_postgres_dag' AIRFLOW_CTX_TASK_ID='submit_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2024-12-30T18:50:53.726705+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-30T18:50:53.726705+00:00'
[2024-12-31T19:06:38.921+0000] {base.py:73} INFO - Using connection ID 'my_spark_conn' for task execution.
[2024-12-31T19:06:38.923+0000] {spark_submit.py:339} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --name arrow-spark --verbose /usr/local/***/include/scripts/kafka_to_postgres_teams.py
[2024-12-31T19:06:38.941+0000] {spark_submit.py:490} INFO - /opt/spark/bin/spark-class: line 71: /usr/lib/jvm/java-11-openjdk-amd64/bin/java: No such file or directory
[2024-12-31T19:06:38.942+0000] {spark_submit.py:490} INFO - /opt/spark/bin/spark-class: line 96: CMD: bad array subscript
[2024-12-31T19:06:39.007+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 157, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    f"Cannot execute: {self._mask_cmd(spark_submit_cmd)}. Error code is: {returncode}."
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --verbose /usr/local/***/include/scripts/kafka_to_postgres_teams.py. Error code is: 1.
[2024-12-31T19:06:39.046+0000] {taskinstance.py:1350} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_kafka_to_postgres_dag, task_id=submit_spark_job, execution_date=20241230T185053, start_date=20241231T190638, end_date=20241231T190639
[2024-12-31T19:06:39.075+0000] {standard_task_runner.py:109} ERROR - Failed to execute job 112 for task submit_spark_job (Cannot execute: spark-submit --master spark://spark-master:7077 --name arrow-spark --verbose /usr/local/***/include/scripts/kafka_to_postgres_teams.py. Error code is: 1.; 56)
[2024-12-31T19:06:39.153+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2024-12-31T19:06:39.239+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
